src/ncurses_def.cpp

```diff
@@ -325,7 +336,14 @@ void catacurses::init_interface()
     set_escdelay( 10 ); // Make Escape actually responsive
     // TODO: error checking
     start_color();
+    imclient = std::make_unique<cataimgui::client>();
     init_colors();
+#if !defined(__CYGWIN__)
+    // ncurses mouse registration
+    mouseinterval( 0 );
+    mousemask( ALL_MOUSE_EVENTS | REPORT_MOUSE_POSITION, nullptr );
+    printf( "\033[?1003h\n" );
```

Comment:
```suggestion
    printf( "\033[?1003h\n" ); // ←[?1003h
```

Will printf eventually be removed? If so, maybe add `//TODO remove` so that people like me get it.

Otherwise, I would comment on what does it print.
I compiled that one line and got this on Windows machine
```
←[?1003h
```

Reply:
```
  printf("\033[?1003h\n"); // Makes the terminal report mouse movement events
```

It's an xterm control code to enable mouse events.

Reply:
FWIW, that is a DECSET, turning on mode 1003 so that the terminal will send all mouse events to the application. Not supported on windows though, or by every terminal emulator in existence. ncurses is supposed to handle that in the mousemask function though, iirc.

Reply:
Yea, `man mousemask` has many details. Perhaps the printf was added to debugging or somesuch?

Reply:
Yeah let me revisit this line. I was trying to add mouse pointer events for ImGui. Maybe it wasn't necessary.

Reply:
finally remembered to look at this. @Brambor @akrieger on Konsole and on the built-in terminal in VS Code, mousemask on its own does not work.... I found the printf in some code examples online. Some people online say that the TERM variable might not be set properly which causes mousemask to not behave? There might be a bigger conversation to be had here, because unfortunately, without mouse position updates, ImTui does not work very well with the mouse at all. ImTui needs a few frames to finish a mouse click operation, and when drawing is blocked by input like in curses builds, not having the mouse position updates is a huge problem.

Reply:
Sorry, just to be clear - mousemask turns on mouse button events, just not mouse position.

Reply:
Maybe put something like this as an explanation into the code, or at least a reference to this conversation.

Reply:
But I didn't know anything, I was just surprised there is a printf 🦆

Reply:
Yep, I agree. I added a comment. Resolving this, re-open it if you want to discuss this more

Reply:
This is still the wrong solution though. It's either redundant, because `mousemask` worked as expected, or it is papering over some other problem that we shouldn’t hide from the user.

On the other hand, I can’t help but notice that ImTui has the same code and presumably for the same reason.

Anyway, shouldn’t this code just go away in favor of calling `ImTui_ImplNcurses_Init`?